crates/outbound-networking/src/lib.rs

```diff
@@ -172,12 +173,19 @@ impl HostConfig {
             bail!("hosts must not contain paths");
         }
 
+        if let Some((subdomain, domain)) = host.split_once('.') {
+            if subdomain == "*" {
+                return Ok(Self::AnySubdomain(format!(".{domain}")));
+            }
```

Comment:
Should be equivalent:
```suggestion
        if let Some(domain) = host.strip_prefix("*.") {
            return Ok(Self::AnySubdomain(format!(".{domain}")));
```